### ISSUE TITLE
Build failure on OSX: undeclared identifier z_const

### ISSUE THREAD

**Issue by durka** [tectonic-typesetting/tectonic#102]
I followed the homebrew/cargo-install instructions. cargo-install stops at:

```
running: "cc" "-O3" "-ffunction-sections" "-fdata-sections" "-fPIC" "-m64" "-I" "." "-I" "/var/folders/7l/cvkn0kbj307fyjc6zkfs6g_w0000gn/T/cargo-install.vXpTqybMO0Q8/release/build/tectonic-65e183164a4968bd/out" "-I" "/usr/local/Cellar/icu4c/58.2/include" "-I" "/usr/local/Cellar/harfbuzz/1.4.6_1/include/harfbuzz" "-I" "/usr/local/Cellar/glib/2.52.2/include/glib-2.0" "-I" "/usr/local/Cellar/glib/2.52.2/lib/glib-2.0/include" "-I" "/usr/local/opt/gettext/include" "-I" "/usr/local/Cellar/pcre/8.40/include" "-I" "/usr/local/opt/freetype/include/freetype2" "-I" "/usr/local/Cellar/graphite2/1.3.10/include" "-I" "/usr/local/Cellar/libpng/1.6.29/include/libpng16" "-DHAVE_GETENV=1" "-DHAVE_LIBPNG=1" "-DHAVE_MKSTEMP=1" "-DHAVE_STDINT_H=1" "-DHAVE_SYS_TYPES_H=1" "-DHAVE_SYS_WAIT_H=1" "-DHAVE_TM_GMTOFF=1" "-DHAVE_ZLIB=1" "-DHAVE_ZLIB_COMPRESS2=1" "-DXETEX_MAC=1" "-o" "/var/folders/7l/cvkn0kbj307fyjc6zkfs6g_w0000gn/T/cargo-install.vXpTqybMO0Q8/release/build/tectonic-65e183164a4968bd/out/tectonic/dpx-pdfobj.o" "-c" "tectonic/dpx-pdfobj.c"
cargo:warning=tectonic/dpx-pdfobj.c:387:68: warning: format specifies type 'unsigned short' but the argument has type 'unsigned char' [-Wformat]
cargo:warning=            _tt_abort("object type %hu not allowed in xref table", type);
cargo:warning=                                   ~~~                             ^~~~
cargo:warning=                                   %hhu
cargo:warning=tectonic/dpx-pdfobj.c:495:64: warning: format specifies type 'long' but the argument has type 'int' [-Wformat]
cargo:warning=                dpx_message("Compression saved %ld bytes%s\n", compression_saved,
cargo:warning=                                               ~~~             ^~~~~~~~~~~~~~~~~
cargo:warning=                                               %d
cargo:warning=tectonic/dpx-pdfobj.c:1168:65: warning: data argument not used by format string [-Wformat-extra-args]
cargo:warning=                dpx_warning("PDF array element #ld undefined.", i);
cargo:warning=                            ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~  ^
cargo:warning=tectonic/dpx-pdfobj.c:2032:19: error: use of undeclared identifier 'z_const'; did you mean 'const'?
cargo:warning=    z.next_in  = (z_const Bytef *) data; z.avail_in  = len;
cargo:warning=                  ^
cargo:warning=tectonic/dpx-pdfobj.c:2305:19: error: use of undeclared identifier 'z_const'; did you mean 'const'?
cargo:warning=    z.next_in  = (z_const Bytef *) data; z.avail_in  = len;
cargo:warning=                  ^
cargo:warning=tectonic/dpx-pdfobj.c:2692:35: warning: passing 'char []' to parameter of type 'unsigned char *' converts between pointers to integer types with different sign [-Wpointer-sign]
cargo:warning=        ttstub_input_read(handle, work_buffer, n);
cargo:warning=                                  ^~~~~~~~~~~
cargo:warning=./tectonic/core-bridge.h:155:78: note: passing argument to parameter 'data' here
cargo:warning=extern ssize_t ttstub_input_read (rust_input_handle_t handle, unsigned char *data, size_t len);
cargo:warning=                                                                             ^
cargo:warning=tectonic/dpx-pdfobj.c:2735:43: warning: passing 'char []' to parameter of type 'unsigned char *' converts between pointers to integer types with different sign [-Wpointer-sign]
cargo:warning=    nread = ttstub_input_read(pf->handle, work_buffer, nmax);
cargo:warning=                                          ^~~~~~~~~~~
cargo:warning=./tectonic/core-bridge.h:155:78: note: passing argument to parameter 'data' here
cargo:warning=extern ssize_t ttstub_input_read (rust_input_handle_t handle, unsigned char *data, size_t len);
cargo:warning=                                                                             ^
cargo:warning=tectonic/dpx-pdfobj.c:2815:35: warning: passing 'char *' to parameter of type 'unsigned char *' converts between pointers to integer types with different sign [-Wpointer-sign]
cargo:warning=    ttstub_input_read(pf->handle, buffer, length);
cargo:warning=                                  ^~~~~~
cargo:warning=./tectonic/core-bridge.h:155:78: note: passing argument to parameter 'data' here
cargo:warning=extern ssize_t ttstub_input_read (rust_input_handle_t handle, unsigned char *data, size_t len);
cargo:warning=                                                                             ^
cargo:warning=tectonic/dpx-pdfobj.c:2968:14: warning: format specifies type 'unsigned long' but the argument has type 'unsigned int' [-Wformat]
cargo:warning=             obj_num, obj_gen);
cargo:warning=             ^~~~~~~
cargo:warning=tectonic/dpx-pdfobj.c:3284:87: warning: format specifies type 'unsigned long' but the argument has type 'int' [-Wformat]
cargo:warning=                dpx_warning("Invalid xref table entry [%lu]. PDF file is corrupt...", i);
cargo:warning=                                                       ~~~                            ^
cargo:warning=                                                       %d
cargo:warning=tectonic/dpx-pdfobj.c:3672:35: warning: passing 'char [10]' to parameter of type 'unsigned char *' converts between pointers to integer types with different sign [-Wpointer-sign]
cargo:warning=    if (ttstub_input_read(handle, buffer, sizeof(buffer) - 1) != sizeof(buffer) - 1)
cargo:warning=                                  ^~~~~~
cargo:warning=./tectonic/core-bridge.h:155:78: note: passing argument to parameter 'data' here
cargo:warning=extern ssize_t ttstub_input_read (rust_input_handle_t handle, unsigned char *data, size_t len);
cargo:warning=                                                                             ^
cargo:warning=tectonic/dpx-pdfobj.c:3727:53: warning: format specifies type 'unsigned long' but the argument has type 'unsigned int' [-Wformat]
cargo:warning=        dpx_warning("Can't resolve object: %lu %u", obj_num, obj_gen);
cargo:warning=                                           ~~~      ^~~~~~~
cargo:warning=                                           %u
cargo:warning=tectonic/dpx-pdfobj.c:3740:58: warning: format specifies type 'unsigned long' but the argument has type 'unsigned int' [-Wformat]
cargo:warning=            dpx_warning("Could not read object: %lu %u", obj_num, obj_gen);
cargo:warning=                                                ~~~      ^~~~~~~
cargo:warning=                                                %u
cargo:warning=11 warnings and 2 errors generated.
```

**Comment by rekka** [comment]
It looks like it's picking a wrong version of `zlib.h`. I can't see it's include path in the compiler flags. Could you post the output of the following two commands?

```
pkg-config --libs --cflags  zlib
pkg-config --modversion zlib
```

**Comment by durka** [comment]
Within `brew sh`, I get:

```
brew ~$ pkg-config --libs --cflags zlib
-I/usr/local/Cellar/zlib/1.2.11/include -L/usr/local/Cellar/zlib/1.2.11/lib -lz
brew ~$ pkg-config --modversion zlib
1.2.11
```

**Comment by rekka** [comment]
Thanks! That looks fine, indeed. But I would expect to see the path `/usr/local/Cellar/zlib/1.2.11/include` in the invocation of `cc` in your first post, since the build script just invokes `pkg-config` to get these paths. Have you run `cargo install` within `brew sh`? If not, try the `pkg-config` commands in the same environment in which `cargo install` failed.

**Comment by durka** [comment]
I'll try the cargo install in the brew shell but I can't get back to it
until Sunday. Will keep you posted!

**Comment by pkgw** [comment]
This may have been fixed by #120. @durka, if you're able to reproduce the problem and are set up to try building the unreleased master branch, could you let us know? Thanks.

**Comment by durka** [comment]
Yes, the master branch compiles!
